File: backend/LabsAndCourses.Domain/Course.cs

namespace LabsAndCoursesManagement.Domain
{
    public class Course
    {
        public Course( string title, int semester, int credits)
        {
            Id = Guid.NewGuid();
            Title = title;
            Semester = semester;
            Credits = credits;
        }

    
        public Guid Id { get; private set; }
        public string Title { get; private set; }
        public int Semester { get; private set; }
        public int Credits { get; private set; }

        // update
        public void Update(Course updatedCourse)
        {
            Title = updatedCourse.Title;
            Semester = updatedCourse.Semester;
            Credits = updatedCourse.Credits;
        }

    }
}

File: frontend/src/entities/Course.ts
import { Entity, Of, createEntityStore } from 'entity-of';


@Entity
export class Course {
  @Of(() => String, { optional: true })
  id = ''
  @Of(() => String)
  title = ''
  @Of(() => Number)
  semester = 0
  @Of(() => Number)
  credits = 0

  static of = Entity.of<Course>();
}
